projects/packages/waf/run.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * This file is to be included where the Jetpack Waf is to be run. Note that it will potentially stop the whole
+ * request as this is the point of a functioning firewall.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+namespace Automattic\Jetpack\Waf;
+
+require_once __DIR__ . '/vendor/autoload.php';
```

Comment:
I think so; because if we run in standalone mode, the entry point from a PHP perspective is `bootstrap.php` and then this file and not any other file where "composer is already happening anyway". See also [this diagram](https://github.com/Automattic/jetpack/pull/23076#discussion_r817736336) for reference.

Reply:
An alternative may be to move this `require_once` line into the `bootstrap.php` file if we want to avoid the redundant call.

Reply:
@bhoop True, but I found it easier to argue about when it is here, so we are "safe from all sides" when the run.php is loaded (where the autoloading is also important to work). If we have it in the bootstrap.php, it is not technically required in there since no classes are called, so one would kind of split up the problem (require the autoloading in the bootstrap.php, so it can be used in the run.php here). 

Hence, I found it easier to keep the things together that belong together. Does that make sense?

Reply:
@anomiex Thanks for looking into this! Do you happen to have a suggestion on how to resolve this? The thing is that we do need to activate the autoloader here to have the classes that the package uses available when running in standalone mode (meaning, before WordPress even started). We could of course only require specific files from our package, but It seems that this would leave us with very much the same problem.

Is there a Jetpack-default way to say "take the path from the plugin" or something like that? Then again, we may not even be able to use that since we are running before WordPress here. It appears to me that traversing directories up until I find a `vendor/autoload.php` would not be a particularly safe option. I could however potentially check both places and load the one that I find. Does that sound like it would make sense at all?

Reply:
I'd probably do as bhoop suggested and put it in the bootstrap file. When you generate that file, you should be able to derive the proper path to `autoload.php` to embed in there.

You could try `/../../../vendor/autoload.php` if you insist, but that might fail in dev environments where stuff is symlinked.

Reply:
@anomiex Thank you again for your insight on this. I've now changed how the autoloader is required as discussed. I also found that the case of the `jetpack-waf` directory not yet existing was not really covered, so I added a check for that; and covered both these changes with tests. @jeherve If one of you could take another look here and let me know if there are more changes required, that would be highly appreciated. Thanks! :)